mr/src/main/java/org/opensearch/hadoop/rest/commonshttp/auth/aws/AwsV4Signer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.hadoop.rest.commonshttp.auth.aws;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.opensearch.hadoop.cfg.Settings;
+import org.opensearch.hadoop.rest.Request;
+import org.opensearch.hadoop.thirdparty.google.common.base.Optional;
+import org.opensearch.hadoop.thirdparty.google.common.base.Splitter;
+import org.opensearch.hadoop.thirdparty.google.common.base.Strings;
+import org.opensearch.hadoop.thirdparty.google.common.base.Supplier;
+import org.opensearch.hadoop.thirdparty.google.common.collect.ImmutableListMultimap;
+import org.opensearch.hadoop.thirdparty.google.common.collect.Multimap;
+import org.opensearch.hadoop.thirdparty.google.common.collect.ImmutableMap;
+import org.opensearch.hadoop.thirdparty.google.common.collect.ImmutableList;
+import org.opensearch.hadoop.thirdparty.google.common.base.Joiner;
+import org.opensearch.hadoop.thirdparty.apache.commons.httpclient.HttpMethod;
+
+import com.amazonaws.DefaultRequest;
+import com.amazonaws.auth.AWS4Signer;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSCredentialsProviderChain;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.http.HttpMethodName;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.util.SdkHttpUtils;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.TreeMap;
+import org.opensearch.hadoop.util.ByteSequence;
+import java.util.List;
+import java.util.Map;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.URLEncoder;
+
+public class AwsV4Signer {
```

Comment:
Could you rename this class to something like `AWS4SignerSupport` or alike, it leads to confusion with `com.amazonaws.auth.AWS4Signer`